run

```diff
@@ -885,38 +893,50 @@ for k, v in obj[project_type]["run"].items():
 #===========================================================================================
 # Other options
 
+list_cmake_dir_options() {
+  cmake_function=$1
+  cmakelists=$(find ${SCRIPT_DIR} -name "CMakeLists.txt")
+  cmake_add_dir=$(grep --no-filename $cmake_function $cmakelists)
+  echo "$cmake_add_dir" | sed -n "s|$cmake_function([[:space:]]*\([^[:space:])]*\).*|\1|p" | sort
+}
```

Comment:
Nice! Parsing for macro calls seems reasonable under existing infrastructure.

Reply:
I think it might actually be more robust, since these macro calls are our ground truth to what can really be built.